src/Workspaces/CSharp/Portable/Editing/CSharpImportAdder.cs

```diff
@@ -66,138 +64,92 @@ protected override SyntaxNode MakeSafeToAddNamespaces(SyntaxNode root, IEnumerab
             return null;
         }
 
-        private class Rewriter : CSharpSyntaxRewriter
+        private class ConflictWalker : CSharpSyntaxWalker
```

Comment:
this file is likely easier to review side-by-side.  the final impl is effectively all new.